networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,303 @@
+    connected, and can be applied to networks with many components.
+
+    Parameters
+    ----------
+    G : graph
+        A NetworkX graph.
+
+    analytical: bool, optional (default=False)
+        Whether the analytical or iterative formulation
+        for computing DomiRank should be used.
+
+    alpha: float, optional (default=0.95)
+        The level of competition for DomiRank.
+
+    dt: float, optional (default=0.1)
+        The step size for the Newton iteration.
+
+    epsilon: float, optional (default=1e-5)
+        The relative stopping criterion for convergence.
+
+    max_iter: integer, optional (default=50)
+        Maximum number of Newton iterations allowed.
+        It is recommended that ''max_iter >= 50''.
+
+    patience: integer, optional (default=10)
+        The number of steps between convergence checks.
+        It is recommended that ''patience >= 10''.
+
+
+    Returns
+    -------
+    nodes : dictionary
+        Dictionary keyed by node with DomiRank centrality of the node as value.
+
+    sigma : float
+        $\alpha$ normalized by the smallest eigenvalue.
+
+    converged: boolean
+        Whether the centrality computation converged.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(5)
+    >>> centrality, sigma, converged = nx.domirank(G)
+    >>> print([f"{node} {centrality[node]:0.2f}" for node in centrality])
+    ['0 -0.54', '1 1.98', '2 -1.08', '3 1.98', '4 -0.54']
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If the graph `G` is the null graph.
+
+    NetworkXUnfeasible
+        If `alpha` is negative (and thus outside its bounds): ``alpha < 0``.
+
+    NetworkXUnfeasible
+        If ``alpha > 1`` with the ``analytical=False`` argument.
+
+    Warning
+        If supercharging the competition parameter for the analytical solution: ``alpha > 1``.
+
+    Warning
+        If one is using the analytical solution for a large system, i.e. ``len(G) > 5000``, as the algorithm will be slow.
+
+    See Also
+    --------
+    :func:`~networkx.algorithms.centrality.degree_centrality`
+    :func:`~networkx.algorithms.link_analysis.pagerank_alg.pagerank`
+    :func:`~networkx.algorithms.link_analysis.hits_alg.hits`
+
+    References
+    ----------
+    .. [1] Engsig, M., Tejedor, A., Moreno, Y. et al.
+        "DomiRank Centrality reveals structural fragility of complex networks via node dominance."
+        Nat Commun 15, 56 (2024). https://doi.org/10.1038/s41467-023-44257-0
+    """
+    import numpy as np
+    import scipy as sp
+
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept(
+            "cannot compute centrality for the null graph"
+        )
+    GAdj = nx.to_scipy_sparse_array(G)  # convert to scipy sparse csr array
+
+    # Here we renormalize alpha with the smallest eigenvalue (most negative eigenvalue) by calling the "hidden" function _find_smallest_eigenvalue()
+    # Note, this function always uses the recursive definition
+    if analytical == False:
```

Comment:
I don't know what the standard is or should be in the codebase, but I personally find `if not analytical` more readable. It does behave differently though! Consider which block gets executed if for whatever reason `analytical=None`.

(@dschult, do you have a strong opinion on this? Is there a standard practice in the NetworkX codebase?)